brave/brave-core
# feat(wallet): Max Slippage Modal

## Description 

- Introduces a new `Max Slippage` modal to allow user to set the `Slippage Tolerance` for a `Swap` or `Bridge` transaction.
- Re-sorts the `Quote Info` to give `Slippage` higher priority in the display.


Resolves <https://github.com/brave/brave-browser/issues/39514>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Open the `Wallet` and navigate to the `Swap` or `Bridge` screen.
2. Select some tokens and get a quote.
3. Once the `Quote Info` appears click on the `Max slippage` button
4. Test changing between `Suggested` and `Custom` slippage.
5. After you update the `slippage`, the quote should automatically be refreshed and reflect the `Slippage` amount you entered.

https://github.com/brave/brave-core/assets/40611140/072bc847-32a8-4bf9-875f-ccb0cf99b5f1
